libcaf_test/README.md

```diff
@@ -169,6 +169,51 @@ will see the vector with its two initial elements. However, since `AND_*` blocks
 run after the previous block, the `AND_WHEN` block in the example above will see
 the vector with three elements.
 
+Scenario Outlines
+-----------------
+
+Scenario outlines allow users to run the same scenario with different inputs.
+The `GIVEN`, `WHEN`, `THEN`, etc. blocks inside of a scenario outline can use
+Gherkin-style placeholders to refer to input values using the syntax `<name>`.
+Further, an outline must contain an `EXAMPLE` block that defines the input
+values in Markdown table notation.
+
+To retrieve the values(s) for the placeholder(s) in a block, users can call
+`block_parameters<...>()` with `...` replaced by the desired type(s).
+
+The following example generates two scenarios from the outline. The first
+scenario will have `start = 12`, `eat = 5`, and `left = 7`. The second scenario
+will have `start = 20`, `eat = 5`, and `left = 15`.
+
+```cpp
+OUTLINE("eating cucumbers") {
+  GIVEN("there are <start> cucumbers") {
+    auto start = block_parameters<int>();
+    auto cucumbers = start;
+    WHEN("I eat <eat> cucumbers") {
+      auto eat = block_parameters<int>();
```

Comment:
It's just:

```
WHEN("I have <foo> and <bar>") {
  auto [foo, bar] = block_parameters<int, int>();
```

But I'll add an example for it since it's not really mentioned anywhere.